### PR TITLE
Multiple issues on workspace summary tab (no JIRA)

### PR DESCRIPTION
**Pull request by MattPutnam**
1. The summary tab, like all tabs, requests a workspace refresh upon loading. This bubbles up to a containing component, then trickles back down as a props change. Previously, the workspace tab component was too aggressively refreshing when props changed. The result was that if you clicked "Edit" immediately as it became available, the refresh could reload the page. Selenium tests do exactly this, so they were failing sporadically on tests that edit things.
2. The tests regarding the rendering of links in the workspace attributes table were broken. The dos links resolve before creating a popup, so there needed to be a wait in that case.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by MattPutnam** [comment]
jenkins retest

**Comment by dvoet** [comment]
please run tests on this several times before merging

**Comment by UrsaStutsman** [comment]
This looks good to me. But definitely should do multiple runs as Doug suggested.

**Comment by aweng98** [comment]
I recommend additional changes that I'd like to include in this PR.

Editing `WorkspaceMethodConfigDetailsPage.scala`

``` 
private val cancelEditedButton = Button("cancel-edit-method-config-button")

def openEditMode(expectSuccess: Boolean = true): Unit = {
    openEditModeButton.doClick()
    if (expectSuccess) {
      saveEditedMethodConfigButton.awaitEnabledState()
      cancelEditedButton.awaitEnabledState()
      saveEditedMethodConfigButton.awaitVisible()
    }
  }

  def saveEdits(expectSuccess: Boolean = true): Unit = {
    // The button can sometimes scroll off the page and become unclickable. Therefore we need to scroll it into view.
    saveEditedMethodConfigButton.scrollToVisible()
    saveEditedMethodConfigButton.doClick()
    if (expectSuccess) {
      openEditModeButton.awaitEnabledState()
      openLaunchAnalysisModalButton.awaitVisible()
    }
  }

```

**Comment by MattPutnam** [comment]
1 success. Jenkins retest

**Comment by MattPutnam** [comment]
fiab-start failed. Jenkins retest

**Comment by MattPutnam** [comment]
Legit test failure

**Comment by MattPutnam** [comment]
fiab-start failed. Jenkins retest

**Comment by MattPutnam** [comment]
fiab-start failed. Jenkins retest

**Comment by MattPutnam** [comment]
1 success (since new commit). Jenkins retest

**Comment by MattPutnam** [comment]
2 failures, seemingly unrelated. 1 billing creation 400, 1 failure to click the "Launch analysis" button.

Jenkins retest

**Review by dvoet — CHANGES_REQUESTED** [review]
I don't see how this fixes the problem. And the last test failed with a "Tried to add workspace attribute while not editing" error too.

**Comment by MattPutnam** [comment]
fiab-start failure

jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala [hunk 1]
```diff
@@ -233,7 +153,7 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
     await ready new CloneWorkspaceModal
   }
 
-  private def isEditing: Boolean = {
+  private[summary] def isEditing: Boolean = {
     !sidebar.editButton.isVisible
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala]: It'd be a better verification if this is checking for `sidebar.cancelButton` and `saveButton` visible. Correctly verifying expected UI elements displayed.

**Comment by MattPutnam** [comment]
2 unrelated failures, submissions didn't complete

**Comment by dvoet** [comment]
@MattPutnam can you please break out the most recent set of fixes into its own PR. I am not convinced the rest of it is worth doing given it didn't actually solve the problem. If it ends up being a good refactor, great, we can roll it out after the freeze is lifted. But for now let's keep changes to a minimum.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala [hunk 2]
```diff
@@ -282,9 +201,7 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
     action
 
     if (isEditing) {
-      sidebar.saveButton.scrollToVisible()
```
Comment (MattPutnam) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala:285]: Clickable.doClick() already scrolls to visible

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala [hunk 3]
```diff
@@ -282,9 +201,7 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
...
       sidebar.clickSave()
-      awaitReady()
```
Comment (MattPutnam) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala:287]: clickSave() already calls awaitReady()

**Comment by MattPutnam** [comment]
@dvoet One of the reasons we're in this hell is because it's too easy to write broken tests. My refactors are all about making it easier to write correct test code. All of these changes are about making the code simpler and more correct at the same time. If I back out the refactors we're just making this codebase worse.

**Comment by aweng98** [comment]
I think Matt is on right path. Tests are easily broken mainly due to brittle Page and framework methods. Brittle methods need to be fixed.

**Comment by MattPutnam** [comment]
1 PreviewSpec failure that seems to be expected.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala [hunk 4]
```diff
@@ -298,3 +215,87 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
...
+    newButton.doClick()
+    awaitReady()
+  }
+
+  def clickForPreviewPane(link: String): GCSFilePreviewModal = {
+    click on LinkTextQuery(link)
+    val previewModal = new GCSFilePreviewModal()
+    // dos links take a sec to pop up
+    previewModal.awaitVisible()
+    await ready previewModal
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala:293]: style-nit: parentheses are used every where, but not here in `previewModal`

**Review by aweng98 — APPROVED** [review]
I don't find anything wrong with test code. I can't comment on UI react code because don't understand them. This PR looks good to me.

**Comment by aweng98** [comment]
if I may to make a suggestion, can you consider making enhancement to `Collapse.ensureCollapsed` and `ensureExpanded` methods to something similar like this. Let me know if you have question about why.
```
  def ensureCollapsed2(): Unit = {
    await condition {
      awaitReady()
      click on toggleComponent
      stateOf(toggleComponent) == "collapsed"
      // maybe a very very small sleep here
    }
  }
```

**Review by dvoet — APPROVED** [review]

**Pull request merged**